climpred/classes.py

```diff
@@ -350,7 +351,30 @@ def _apply_xr_func(v, name, *args, **kwargs):
                 #            as ds[dim] and the dim doesn't exist as a key.
                 # DimensionError: This accounts for our custom error when applying
                 # some stats functions.
-                except (ValueError, KeyError, DimensionError):
+                except (ValueError, KeyError, DimensionError) as e:
+                    if args == tuple():
+                        func_name = None
+                    else:
+                        if callable(args[0]):
+                            func_name = args[0].__name__
+                        else:
+                            func_name = "not a callable"
+                    dim = kwargs.get("dim", False)
+                    error_type = type(e).__name__
+                    if len(args) > 1:
+                        msg = f"{func_name}({args[1:]}, {kwargs}) 'failed\n{error_type}:{e}"
+                    else:
+                        msg = f"{func_name}({kwargs}) 'failed\n{error_type}:{e}"
+                    if set(["lead", "init"]).issubset(set(v.dims)):  # initialized
```

Comment:
Similarly here. I don't know what you mean by just saying `# initialized`

Reply:
I guess maybe say `# error due to issue with initialized forecast`

Reply:
ok. yep. thats what I tried to do here.